4.0/en/0x21-V13-API.md

```diff
@@ -56,6 +56,18 @@ Note: Due to issues with XXE attacks against DTDs, DTD validation should not be
 | **13.4.1** | Verify that a query allow list or a combination of depth limiting and amount limiting is used to prevent GraphQL or data layer expression Denial of Service (DoS) as a result of expensive, nested queries. For more advanced scenarios, query cost analysis should be used. |  | ✓ | ✓ | 770 |
 | **13.4.2** | Verify that GraphQL or other data layer authorization logic should be implemented at the business logic layer instead of the GraphQL layer. |  | ✓ | ✓ | 285 |
 
+## V13.5 WebSocket Security Requirements
+
+| # | Description | L1 | L2 | L3 | CWE |
+| :---: | :--- | :---: | :---:| :---: | :---: |
+| **13.5.1** | Verify that only authenticated users can open WebSocket connections. | ✓ | ✓ | ✓ | 306 |
+| **13.5.2** | Verify that WebSocket Secure (wss) is used for all WebSocket connections. | ✓ | ✓ | ✓ | 319 |
+| **13.5.3** | Verify that the Origin header is checked in the initial HTTP WebSocket handshake. | ✓ | ✓ | ✓ | 346 |
+| **13.5.4** | Verify that WebSocket messages are properly sanitized and encoded server-side. | ✓ | ✓ | ✓ | 116 |
+| **13.5.5** | Verify that rate limiting is in place for WebSocket messages | ✓ | ✓ | ✓ | 770 |
+| **13.5.6** | If you are using session or channel tokens in webSockets, verify that the tokens are obtained from a response to secure POST request only, and are not sent by the server through the Websocket connection. | ✓ | ✓ | ✓ | 319 |
+| **13.5.7** | If you are using session or channel tokens in webSockets, verify that the tokens possess at least 64 bits of entropy and are generated using approved cryptographic algorithms. | ✓ | ✓ | ✓ | 331 |
```

Comment:
Would it be better to reference the session token guidance contained elsewhere in this standard? Are these requirements different to other session token requirements which we already have?

Reply:
You are right. This part is a repetition of section 3.2 of the standard. However, websocket session tokens are often different than cookie session tokens. They should be **checked separately**.

What do you think would be the best solution?

- Reference section 3.2 in the 13.5.7 requirement?
    - Such as done in rule 3.2.3. It makes the requirement explicit, but still creates a bit of repetition
- Replace 13.5.7 by a sentence in the 13.5 section text?
    - It removes the repetition, but creates a "hidden" requirement in the paragraph text. As such we are not sure that this specific token will be checked.